webapp/tests/conftest.py

```diff
@@ -2,3 +2,58 @@
 import sys
 sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..'))
 os.environ["FLASK_ENV"] = 'testing'
+
+import pytest
+
+from app.app import create_app
+from app.extensions import db as _db
+
+
+@pytest.fixture(scope="session")
+def app():
+    _app = create_app()
+
+    with _app.app_context():
+        yield _app
+
+
+@pytest.fixture
+def client(app):
+    yield app.test_client()
+
+
+@pytest.fixture
+def test_request_context(app):
+    with app.test_request_context() as req:
+        yield req
+
+
+@pytest.fixture(scope="session")
+def db(app):
+    """Ensure tables are created and dropped at the start and end of test runs."""
+    with app.app_context():
+        _db.create_all()
+
+    yield _db
+
+    # Explicitly close DB connection
+    _db.session.close()
+    _db.drop_all()
+
+
+def _truncate_all_tables(db):
+    tables = db.metadata.sorted_tables
+    connection = db.engine.connect()
+    transaction = connection.begin()
+    connection.execute('PRAGMA foreign_keys = OFF;')  # SQLite specific
```

Comment:
If this is SQLite specific, this only works as long as our tests are performed with an SQLite database, correct?

Reply:
Correct.

Reply:
Should we ever move to testing in Postgres, there's an even nicer way to handle this with nested transactions, but that would require having a Postgres DB set up everywhere we want to run the tests.